Riot/Modules/MatrixKit/Models/Room/MXKRoomDataSourceManager.m

```diff
@@ -138,6 +150,12 @@ - (instancetype)initWithMatrixSession:(MXSession *)matrixSession
             }
             
         }];
+        
+        // Observe late decrypted events, and store rooms ids in memory
+        mxEventDidDecryptNotificationObserver = [[NSNotificationCenter defaultCenter] addObserverForName:kMXEventDidDecryptNotification object:nil queue:[NSOperationQueue mainQueue] usingBlock:^(NSNotification *notif) {
+            MXEvent *decryptedEvent = notif.object;
+            [self->roomDataSourcesToDestroy addObject:decryptedEvent.roomId];
+        }];
```

Comment:
```suggestion
        mxEventDidDecryptNotificationObserver = [NSNotificationCenter.defaultCenter addObserverForName:kMXEventDidDecryptNotification object:nil queue:NSOperationQueue.mainQueue usingBlock:^(NSNotification *notif) {
            MXEvent *decryptedEvent = notif.object;
            [self->roomDataSourcesToDestroy addObject:decryptedEvent.roomId];
        }];
```

Minor: dot notation should work on static vars.

Reply:
I wouldn't care much about that.
Writing more readable code is better to me.
I'll leave this up to you in the end. ;)